### PR TITLE
nf-test to FASTQ_FASTQC_UMITOOLS_TRIMGALORE and pin cutadapt in trimgalore

### PR DESCRIPTION
**Pull request by adamrtalbot**
Early tests for subworfklow FASTQ_FASTQC_UMITOOLS_TRIMGALORE.

I need to flesh out the assert statements a bit more, although this catches the FASTQ files which are the most important.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/tests/main.nf.test.snap [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        "content": [
+            [
+                [
+                    {
+                        "id": "test",
+                        "single_end": false
+                    },
+                    [
+                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
+                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
```
Comment (sateeshperi) [inline subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/tests/main.nf.test.snap:167]: why paths here and no md5 ?

Comment (maxulysse) [reply subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/tests/main.nf.test.snap:167]: that's the input data

Comment (matthdsm) [reply subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/tests/main.nf.test.snap:167]: Why is the input data in the assertions? Shouldn't that be excluded?

**Review by matthdsm — APPROVED** [review]
LGTM 🤷🏻

**Comment by maxulysse** [comment]
Thanks @matthdsm
@sateeshperi if there's any issues, I'll fix in a new PR.
But I'd really to see if this fixes some of my issues in rnaseq, so I'll merge for now.

File: subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/tests/main.nf.test.snap (hunk 1 above)
Comment (maxulysse) [reply subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/tests/main.nf.test.snap:167]: Yeah, this was an issue, I'll fix this...

Comment (maxulysse) [reply subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/tests/main.nf.test.snap:167]: https://github.com/nf-core/modules/pull/5023

**Pull request merged**